### PR TITLE
Python style guide initial starting point

### PR DESCRIPTION
**Pull request by benjaminmcf**
Adding a python style guide to the HQIU-Coding-Styles repository. Provides some general guidance for writing both Python scripts and notebooks. The style guide is based on a combination of the R HQIU coding style guide, tidyverse coding style guide, and PEP 8. 

Submitting the pull request as an opportunity to receive feedback.

I have added Andrew and Peter as reviewers as they are contributors on this repository.

### CODE REVIEW

File: Python/README.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    sentence = f"My name is {}. I am {age} years old"
+    return(sentence)
+
+```
+
+
+### General programming principles
+
+- Keep it simple
+- Dont repeat yourself. Pieces of code should be implemented in just one place in the souce code. Tasks that are repeated should be implemented as functions
```
Comment (pli9) [inline Python/README.md]: Minor spelling mistake for "souce"

File: Python/README.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+
+
+## Notebook usage and best practice
+
+Here are some general usage guidelines and best practices when using jupyter notebooks, databricks notebooks, or similar alternatives. 
+
+- Give the notebook a header. Limit the scope of the notebook to a particular objective so that it does not becoming overwhelming. 
+- Use markdown throughout the notebook. This includes using headings and subheadings where appropriate. Remember that you are documenting this for those that you work for both yourself in the future, and those that you work with. 
+- importing packages and modules should be done at the begining of the notebook only.
```
Comment (pli9) [inline Python/README.md]: NIT: capitalise start of sentence

File: Python/README.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+### General programming principles
+
+- Keep it simple
+- Dont repeat yourself. Pieces of code should be implemented in just one place in the souce code. Tasks that are repeated should be implemented as functions
+- You are not going to need it. Do not implement something until it is necessary. 
+- For more complicated applications or pieces of software, separation of concerns principles should be implemented. Code should be as independent as possible, making it easier to maintain, update, and reuse. 
+
+
+
+## Notebook usage and best practice
```
Comment (pli9) [inline Python/README.md:166]: For Jupyter notebook, would be useful to add a specific note that outputs need to be cleared before committing and pushing to GitHub.

File: Python/README.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+In accordance with the HQIU R style guide.
+
+* We take a disciplined and practical approach to writing code.
+* We regularly check-in code to Github
+* We believe consistency in style is very important.
+* We demonstrate intent explicitly in code, via clear structure and comments where needed.
+
+## Rules
+
+The basis of the style guide is a combination of principles from PEP 8 and the tidyverse style guide, which is a document that provides the guidelines and best practices on writing python code. Using PEP 8 helps to improve the readability of Python Code.
```
Comment (pli9) [inline Python/README.md]: Add link to PEP 8. What are the conflicts of tidyverse style guide and PEP 8? I'm happy to move to PEP 8 if there are inbuilt stylers/linters.

**Review by pli9 — CHANGES_REQUESTED** [review]
Thanks @benjaminmcf. I've added in comments about some small typos, and to consider moving fully to PEP 8 if there are some in-built linters and stylers for simplicity. Happy to discuss.

**Comment by benjaminmcf** [comment]
Thanks for the feedback, Pete. I will make the requested changes and investigate appropriate linters

File: Python/README.md (hunk 1 above)
Comment (benjaminmcf) [reply Python/README.md]: Change has been pushed to this branch

File: Python/README.md (hunk 2 above)
Comment (benjaminmcf) [reply Python/README.md]: Thank you for identifying this.

File: Python/README.md (hunk 3 above)
Comment (benjaminmcf) [reply Python/README.md:166]: A note has been added to this section under a new heading.

File: Python/README.md (hunk 4 above)
Comment (benjaminmcf) [reply Python/README.md]: I have added a suggested formatter and linter. As PEP8 is a guide, not all of the linters and formatters follow it exactly. I have identified 2 options which are both simple, and do not seem to conflict with the tidyverse style guide. Pylint for example checks for snake case, unused variables and unused imports

**Review by benjaminmcf — COMMENTED** [review]
I have made the changes that you have recommended. I also recommend using a combination of the Black formatter and Pylint linter to support writing Python code in a consistent style. Happy to review these options if it they are evaluated as not appropriate.

**Review by benjaminmcf — COMMENTED** [review]
All changes have been made.

**Comment by pli9** [comment]
> I have made the changes that you have recommended. I also recommend using a combination of the Black formatter and Pylint linter to support writing Python code in a consistent style. Happy to review these options if it they are evaluated as not appropriate.

Thanks, it looks good to me. Great that it's part of Anaconda, will make it easy to format and lint. I'll merge it in.

**Pull request merged**